paraatm/simulation_method/vcas.py

```diff
@@ -0,0 +1,380 @@
+
+        Returns
+        -------
+        list
+            include: list[0] trajectory data for reference (right simulation)
+                     list[1::] trajectory data for ignoring each command
+            elements in list is an nparray with 2 dimensions recording timestamp, longitude latitude and altitude
+        """
+        cmd_maintain = self.command_from_file()
+        track = self.simulation()
+        traj = np.asarray([track.time.values.astype(np.float)//1e9, track['latitude'].values,
+                           track['longitude'].values, track['altitude'].values])
+        models = [traj.T]
+        for i in range(len(cmd_maintain)):
+            new_cmd = cmd_maintain[0:i]
+            print(new_cmd)
+            track = self.simulation(new_cmd)
+            traj = np.asarray([track.time.values.astype(np.float) // 1e9, track['latitude'].values,
+                               track['longitude'].values, track['altitude'].values])
+            models.append(traj.T)
+
+        return models
+
+    def model_update(self, *input_prior):
+        """
+        calculate pilot compliance for each command based on results from self.make_anti_model()
+
+        Parameters
+        ----------
+        input_prior : nparray
+            Only takes one nparray storing the pre-existing prior for each command. If none, default is use [0.5, 0.5]
+            as prior
+
+        Returns
+        ----------
+        nparray
+            posterior for obeying each command as time
+        """
+        if len(input_prior) == 0:
+            prior = 0.5 * np.ones((len(self.command_from_file()), 2))
+        else:
+            prior = input_prior[0]
+
+        trajs = self.make_anti_model()
+        ref = trajs[0]
+        compliance = []
+        cmd_timeline = np.append(self.command_from_file()['Timestamp'].values, self.starttime + self.sim_time)
+        for i in range(len(trajs)-1):
+            antimodel = trajs[i+1]
+            obs = self.real[self.real['timestamp'].between(cmd_timeline[i], min((cmd_timeline[i+1]) + 200, cmd_timeline[-1]))]
+            posterior = np.zeros((len(obs) + 1, 3))
+            posterior[0, 0] = cmd_timeline[i]
+            posterior[0, 1:] = prior[i]
+            for j in range(len(obs)):
+                t = obs['timestamp'].values[j]
+                mean = obs['altitude'].values[j]
+                liklhd = sts.norm(mean, 1000)
+                alt_obey = ref[np.abs(ref[:, 0] - t).argmin(), 3]
+                alt_ingr = antimodel[np.abs(antimodel[:, 0] - t).argmin(), 3]
+
+                l_obey = liklhd.pdf(alt_obey)
+                l_ingr = liklhd.pdf(alt_ingr)
+
+                num = posterior[j, 1:] * np.array([l_obey, l_ingr])
+                posterior[j+1, 1:] = num / np.sum(num)
+                posterior[j+1, 0] = t
+
+            compliance.append(posterior)
+
+        return compliance
+
+if __name__ == '__main__':
```

Comment:
You have example code under the `__main__` section in `vcas.py` that may be duplicated in `vcas_example.py`.  It is better for this to be only in one place.  We don't really have a convention yet in para-atm, so you could use either.